src/components/textbox/textbox.pw.tsx

```diff
@@ -0,0 +1,1244 @@
+      "data-component",
+      "button"
+    );
+    await expect(textbox(page).locator(BUTTON)).toBeVisible();
+  });
+
+  testData.forEach((prefix) => {
+    test(`should render with prefix prop as ${prefix}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<TextboxComponent prefix={prefix} />);
+
+      await expect(textboxPrefix(page)).toHaveText(prefix);
+      await expect(textboxPrefix(page)).toHaveCSS("font-size", "14px");
+      await expect(textboxPrefix(page)).toHaveCSS("font-weight", "900");
+      await expect(textboxPrefix(page)).toHaveCSS("margin-left", "12px");
+    });
+  });
+
+  test("should render with prefix prop whilst align prop is also set to 'right'", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<TextboxComponent prefix="foo" align="right" />);
+
+    await expect(textbox(page)).toHaveCSS("flex-direction", "row");
+  });
+
+  /* This test could be deleted, as its specific test case cannot be properly replicated using Playwright,
+  we can assert it exists in the DOM in the specified DOM structure, but cannot properly test its visibility
+  without added complexity or adding new data tags to `Textbox`s DOM. */
+
+  ([
+    [true, "input", "label"],
+    [false, "label", "input"],
+  ] as [boolean, string, string][]).forEach(
+    ([boolean, firstElement, secondElement]) => {
+      test(`should render with reverse prop set to ${boolean}`, async ({
+        mount,
+        page,
+      }) => {
+        await mount(<TextboxComponent reverse={boolean} />);
+
+        const textboxParentParent = await textbox(page).evaluateHandle(
+          (element: Element) => {
+            return element.parentElement?.parentElement;
+          }
+        );
+
+        const domElement1 = await page.evaluate(
+          ({
+            textboxParentParent: parentDomElement,
+            firstElement: element,
+          }) => {
+            const returnedElement = parentDomElement?.querySelectorAll(element)
+              .length;
+            return returnedElement;
+          },
+          { textboxParentParent, firstElement }
+        );
+
+        const domElement2 = await page.evaluate(
+          ({
+            textboxParentParent: parentDomElement,
+            secondElement: element,
+          }) => {
+            const returnedElement = parentDomElement?.querySelectorAll(element)
+              .length;
+            return returnedElement;
+          },
+          { textboxParentParent, secondElement }
+        );
+
+        expect(domElement1).toBeTruthy();
+        expect(domElement2).toBeTruthy();
+      });
+    }
+  );
+
+  /* This test could be deleted, as its specific test case cannot be properly replicated using Playwright,
+  we can assert it exists in the DOM in the specified DOM structure, but cannot properly test its visibility
+  without added complexity or adding new data tags to `Textbox`s DOM. */
+
+  test("should render with positionedChildren prop", async ({
```

Comment:
**suggestion**: I could be wrong, but looking at the original Cypress test for this it seems quite different from the `reverse` one above - and I think it's possible to get a reasonable Playwright equivalent here. Again use a single `.evaluate` to get the parent element, and call `querySelectorAll` with the button locator as you're doing here, checking the length is > 0, or do as the original test did and get the child element and check that attribute directly.

Sadly we can't easily check visibility, but it's still probably worth keeping this in imo. (Happy to be disagreed with though - I don't feel that strongly.)

Reply:
Good shout, managed to cut down the entire test into this `    const dataElementExistsInLabelParent = await textbox(page).evaluate((element) => element.parentElement?.querySelectorAll(`[data-component="button"]`)?.length);
    expect(dataElementExistsInLabelParent).toBeGreaterThan(0);`